node/src/components/storage/tests.rs

```diff
@@ -1395,27 +1481,66 @@ fn should_garbage_collect() {
     check(1);
 }
 
+#[test]
+fn can_put_and_get_block() {
+    let mut harness = ComponentHarness::default();
+
+    // Create a random block using the legacy hashing scheme, store and load it.
+    let (block, merkle_tree_hash_activation) = Block::random_v1(&mut harness.rng);
+    let block = Box::new(block);
+
+    let mut storage = storage_fixture(&harness, merkle_tree_hash_activation);
+
+    let was_new = put_block(&mut harness, &mut storage, block.clone());
+    assert!(was_new, "putting block should have returned `true`");
+
+    // Storing the same block again should work, but yield a result of `true`.
+    let was_new_second_time = put_block(&mut harness, &mut storage, block.clone());
+    assert!(
+        was_new_second_time,
+        "storing block the second time should have returned `true`"
+    );
+
+    let response = get_block(&mut harness, &mut storage, *block.hash());
+    assert_eq!(response.as_ref(), Some(&*block));
+
+    // Also ensure we can retrieve just the header.
+    let response = harness.send_request(&mut storage, |responder| {
+        StorageRequest::GetBlockHeader {
+            block_hash: *block.hash(),
+            responder,
+        }
+        .into()
+    });
+
+    assert_eq!(response.as_ref(), Some(block.header()));
+}
+
 #[test]
 fn can_put_and_get_blocks_v2() {
     let num_blocks = 10;
     let mut harness = ComponentHarness::default();
-    let mut storage = storage_fixture(&harness);
 
-    let era_id = harness
-        .rng
-        .gen_range(MERKLE_TREE_HASH_ACTIVATION..MERKLE_TREE_HASH_ACTIVATION + 10)
-        .into();
+    let era_id = harness.rng.gen_range(0..10).into();
+
+    // Ensure the merkle hashing algorithm is used.
```

Comment:
```suggestion
    // Ensure the Merkle hashing algorithm is used.
```

Reply:
Updated (also in few other places) here https://github.com/casper-network/casper-node/pull/2531/commits/15a212035887b1ced40ec3375626d8c251fef585